### PR TITLE
ADR 31 - separate jobs in cloudfoundry pipeline

### PR DESCRIPTION
**Pull request by tlwr**
What
----

Add an ADR for changes to the pipeline

How to review
-------------

Agree with the ADR

Who can review
--------------

Not @tlwr

### CODE REVIEW

File: source/architecture_decision_records/ADR031-services-core-pipeline-separate-jobs.html.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+deployment to the core platform.
+
+
+# Status
+Accepted
+
+
+# Consequences
+
+The pipeline will no longer be fully controlled by the `pipeline-lock` pool.
```
Comment (richardTowers) [inline source/architecture_decision_records/ADR031-services-core-pipeline-separate-jobs.html.md]:
```suggestion
The services we've separated out will deploy as soon as self-update-pipelines changes the tag_filter of their resource. They will also deploy for each new version of paas-cf. This means they will deploy twice (once when the pipeline-lock job finishes and again when post-deploy finishes).
```

File: source/architecture_decision_records/ADR031-services-core-pipeline-separate-jobs.html.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+These services include:
+
+- PaaS Accounts
+- PaaS Admin
+- PaaS Billing
+- PaaS Metrics
+
+Which are core components to our platform, but not to Cloud Foundry.
+
+Currently these services are unnecessarily coupled in a couple of places:
```
Comment (AP-Hunt) [inline source/architecture_decision_records/ADR031-services-core-pipeline-separate-jobs.html.md]: Could you document the negative side-effects of the coupling? I think the context would benefit from further explanation of the problem this ADR is addressing.

Comment (tlwr) [reply source/architecture_decision_records/ADR031-services-core-pipeline-separate-jobs.html.md]: Would something like

> Unnecessarily coupling has resulted in flakey app tests blocking CVE remediation from reaching production.

be what you had in mind?

Comment (AP-Hunt) [reply source/architecture_decision_records/ADR031-services-core-pipeline-separate-jobs.html.md]: Yep, lgtm.

Comment (tlwr) [reply source/architecture_decision_records/ADR031-services-core-pipeline-separate-jobs.html.md]: Added

**Pull request merged**